src/views/ViewAllExplorer.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useRef, useEffect } from 'react';
+import {
+  Animated,
+  StyleSheet,
+  useColorScheme,
+  FlatList,
+  ActivityIndicator,
+} from 'react-native';
+import { DarkTheme, LightTheme } from '../constants/Colors';
+import { DEVICE_HEIGHT } from '../constants/Platform';
+import WalletItem, { ITEM_HEIGHT } from '../components/WalletItem';
+
+import NavHeader from '../components/NavHeader';
+
+interface ViewAllExplorerProps {
+  isLoading: boolean;
+  explorerData: any;
+  onBackPress: () => void;
+  currentWCURI?: string;
+}
+
+function ViewAllExplorer({
+  isLoading,
+  explorerData,
+  onBackPress,
+  currentWCURI,
+}: ViewAllExplorerProps) {
+  const fadeAnim = useRef(new Animated.Value(0)).current;
+  const isDarkMode = useColorScheme() === 'dark';
+
+  useEffect(() => {
+    Animated.timing(fadeAnim, {
+      toValue: 1,
+      duration: 500,
+      useNativeDriver: true,
+    }).start();
+  }, [fadeAnim]);
+
+  return (
+    <Animated.View style={{ opacity: fadeAnim }}>
+      <>
+        <NavHeader title="Connect your Wallet" onBackPress={onBackPress} />
+        {isLoading || !currentWCURI ? (
+          <ActivityIndicator
+            style={styles.loader}
+            color={isDarkMode ? LightTheme.accent : DarkTheme.accent}
+          />
+        ) : (
+          <FlatList
+            data={explorerData || []}
```

Comment:
Nitpick: When flatlist, what error messaging do we do? I believe it just does ActivityLoader. Anything message you think we can add? Or want to just add it later releases as a Quality of Life (QoL) change?

Reply:
FlatList provides a way to show a custom view if it's empty. Nowadays if the data is not loaded, the modal doesnt open